### PR TITLE
Add mountpoint as CRI image filesystem storage identifier.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/kubernetes/kubernetes/issues/57356.

This PR changes CRI to use mountpoint as storage identifier. See https://github.com/kubernetes/kubernetes/issues/57356#issuecomment-363608733.

Note that:
1) This doesn't work with devicemapper for now. Please feel free to propose change for device mapper, we can discuss more about this after this first version is merged. @mrunalp @runcom
2) `mountpoint` is added as new field in `StorageIdentifier` now. After https://github.com/kubernetes/kubernetes/pull/58973 is merged, we can remove the UUID field in `v1alpha2`. 

/cc @yujuhong @feiskyer @yguo0905 @dashpole @mikebrow @abhi @kubernetes/sig-node-api-reviews 

**Release note**:

```release-note
CRI starts using moutpoint as image filesystem identifier instead of UUID.
```

### CODE REVIEW

**Comment by Random-Liu** [comment]
A container runtime implementation is here https://github.com/containerd/cri-containerd/pull/601.

I've manually tested it, and it works. The CI will catch more issue if there is any.

File: pkg/kubelet/apis/cri/v1alpha1/runtime/api.pb.go [hunk 1]
```diff
@@ -3183,6 +3183,9 @@ func (m *UInt64Value) GetValue() uint64 {
 type StorageIdentifier struct {
 	// UUID of the device.
 	Uuid string `protobuf:"bytes,1,opt,name=uuid,proto3" json:"uuid,omitempty"`
```
Comment (dashpole) [inline pkg/kubelet/apis/cri/v1alpha1/runtime/api.pb.go]: somewhat related: do we plan to keep the UUID field?  If so, what is it used for now that we are passing the mount point?

**Review by dashpole — COMMENTED** [review]
question

File: pkg/kubelet/apis/cri/v1alpha1/runtime/api.pb.go (hunk 1 above)
Comment (dashpole) [reply pkg/kubelet/apis/cri/v1alpha1/runtime/api.pb.go]: read the PR description, ignore :)

**Comment by yujuhong** [comment]
Le'ts wait for the version bump PR to land first.

File: pkg/kubelet/cadvisor/cadvisor_unsupported.go [hunk 2]
```diff
@@ -77,6 +77,6 @@ func (cu *cadvisorUnsupported) WatchEvents(request *events.Request) (*events.Eve
 	return nil, unsupportedErr
 }
 
-func (c *cadvisorUnsupported) GetFsInfoByFsUUID(uuid string) (cadvisorapiv2.FsInfo, error) {
+func (c *cadvisorUnsupported) GetDirFsInfo(path string) (cadvisorapiv2.FsInfo, error) {
```
[the remark this answers is not in the document]
Comment (Random-Liu) [reply pkg/kubelet/cadvisor/cadvisor_unsupported.go]: Done.

File: pkg/kubelet/cadvisor/cadvisor_windows.go [hunk 3]
```diff
@@ -77,6 +77,6 @@ func (cu *cadvisorClient) WatchEvents(request *events.Request) (*events.EventCha
 	return &events.EventChannel{}, nil
 }
 
-func (c *cadvisorClient) GetFsInfoByFsUUID(uuid string) (cadvisorapiv2.FsInfo, error) {
+func (c *cadvisorClient) GetDirFsInfo(path string) (cadvisorapiv2.FsInfo, error) {
```
[the remark this answers is not in the document]
Comment (Random-Liu) [reply pkg/kubelet/cadvisor/cadvisor_windows.go]: Done.

**Comment by Random-Liu** [comment]
Rebased and switch to `v1alpha2`.

File: pkg/kubelet/apis/cri/runtime/v1alpha2/api.proto [hunk 4]
```diff
@@ -1105,8 +1105,8 @@ message UInt64Value {
 
 // StorageIdentifier uniquely identify the storage..
 message StorageIdentifier{
-    // UUID of the device.
-    string uuid = 1;
+    // Mountpoint of the image filesystem.
```
Comment (yujuhong) [inline pkg/kubelet/apis/cri/runtime/v1alpha2/api.proto]: The `StorageIdentifier` is a more generic message type. The comment should not include "image filesystem".

On the other hand, is `StorageIdentifier` still an accurate name, or should we rename it to, say, `FilesystemIdentifier`?

Comment (Random-Liu) [reply pkg/kubelet/apis/cri/runtime/v1alpha2/api.proto]: Will do then.

Comment (Random-Liu) [reply pkg/kubelet/apis/cri/runtime/v1alpha2/api.proto]: Done. Changed to `FilesystemIdentifier`.

**Comment by mrunalp** [comment]
@sameo PTAL.

**Comment by mrunalp** [comment]
:+1: We can add pool name in a follow-on for device mapper.

**Comment by dashpole** [comment]
Code changes lgtm.  I also agree with the idea of using mountpoint as an identifier.

**Pull request merged**